src/components/scene/device-motion-permission-ui.js

```diff
@@ -0,0 +1,161 @@
+
+var DEVICE_PERMISSION_FULL_CLASS = 'a-device-motion-permission-full';
+var DEVICE_PERMISSION_FULL_CENTER_CLASS = 'a-device-motion-permission-full-center';
+var DEVICE_PERMISSION_ACCEPT_CLASS = 'a-device-motion-permission-accept';
+var DEVICE_PERMISSION_CANCEL_CLASS = 'a-device-motion-permission-cancel';
+var DEVICE_PERMISSION_REQUEST_CLASS = 'a-device-permision-request';
+
+/**
+ * UI for enabling device motion permission
+ */
+module.exports.Component = registerComponent('device-motion-permission-ui', {
+  schema: {
+    enabled: { default: true },
+    orientationChange: {default: null},
+    deviceOrientation: {default: null},
+    deviceMotionEl: { default: '' }
+  },
+  init: function () {
+    this.deviceMotionEl = null;
+    this.onDeviceMotionClick = bind(this.onDeviceMotionClick, this);
+    this.onOrientationChangeClick = bind(this.onOrientationChangeClick, this);
+    this.grantedDeviceMotion = bind(this.grantedDeviceMotion, this);
+    if (typeof window.orientation !== 'undefined') {
+      try {
+        if (
+          DeviceOrientationEvent &&
+          typeof DeviceOrientationEvent.requestPermission === 'function'
+        ) {
+          DeviceOrientationEvent.requestPermission()
+            .then(response => {
+              if (response === 'granted') {
+                this.grantedDeviceMotion();
+              }
+            })
+            .catch(err => {
+              console.log(err);
+              this.deviceMotionEl = createDeviceMotionPermissionWindow(
+                this.onDeviceMotionClick,
+                this
+              );
+              this.el.appendChild(this.deviceMotionEl);
+            });
+        } else {
+          this.grantedDeviceMotion();
+        }
+      } catch (oops) {
+        this.grantedDeviceMotion();
+      }
+    } else {
+      this.remove();
+    }
+  },
+
+  remove: function () {
+    if (this.deviceMotionEl) {
+      this.el.removeChild(this.deviceMotionEl);
+    }
+  },
+
+  /**
+   * Enable device motion permission when clicked.
+   */
+  onDeviceMotionClick: function () {
+    try {
+      if (
+        DeviceOrientationEvent &&
+        typeof DeviceOrientationEvent.requestPermission === 'function'
+      ) {
+        DeviceOrientationEvent.requestPermission()
+          .then(response => {
+            if (response === 'granted') {
+              this.grantedDeviceMotion();
+            } else {
+              console.log('Device Motion permission not granted.');
+            }
+          })
+          .catch(console.error);
+      } else {
+        this.grantedDeviceMotion();
+      }
+    } catch (oops) {
+      console.log(
+        'Your device and application combination do not support device motion events.'
+      );
+    }
+  },
+
+  grantedDeviceMotion: function () {
+    const orientationChange=this.el.getAttribute('orientation-change');
+    const deviceOrientation=this.el.getAttribute('device-orientation');
+    this.remove();
+      if(orientationChange){
```

Comment:
We probably don't need handling orientation changes here

Reply:
Agreed--before DeviceOrientationChange was included as a global, I was working around eslint errors. I'll include this in the next push.

Reply:
Sorry, there must be something I'm not understanding. Why does orientation changes relate to the dialog? Is not using CSS sufficient? What are the attributes above for?

Reply:
What is the grantedDeviceMotion method for? It seems to be only handling orientation changes that are already handled by A-Frame.

Reply:
@dmarcos, grantedDeviceMotion unifies Android/iOS device orientation and orientation change methods. For a developer, it gives one entry point to ensure that event triggers are able to be handled in like fashion. It may be a moot issue for non-iOS 13 device/browsers combinations; it is a place where aframe events can be added confidently. The biggest use case is, if someone attempts to include Device Orientation behavior before permission is given, it might fail without the user knowing that it has failed. It provides a single place where events can be associated with user interaction; a place where aframe events may be attached with confidence.

Reply:
Sorry, I'm still missing something. Could you help me understand by describing a concrete use case scenario? If there's an issue related to `orientationchange` events I feel it should be handled somewhere else. Regarding the solution, using global variable names passed through HTML attributes not associated to any component doesn't feel consistent with rest of A-Frame. I made a few cosmetic changes to the PR and removed for now. We can reintroduce if necessary.

Reply:
If the issue is notifying an application if DeviceOrientationEvent.requestPermission has succeeded or failed component events would be the way to go. e.g: `deviceorientationeventsgranted` / `deviceorientationeventsdenied`. Would that work?

Reply:
Both orientation change and device orientation are granted permission by the same event, DeviceMotionEvent. That is why I included them both. Regarding the global variable name; I completely agree with you. I would think that there might be a way to pass the functions that depend on access to device motion to grantedDeviceMotion. That way, it is ensured that iOS/iPad will not have issues associated with attempting to include functionality before they have permission. One way or another, and it could be deviceorientationeventsgranted / deviceorientationeventsdenied, the developer should have access to a flag to know that it is possible to add a deviceorientation or orientationchange event. Thanks, and I am glad that I have been able to contribute to something that I needed to work! :)